mungegithub/mungers/submit-queue.go

```diff
@@ -1070,6 +1077,17 @@ func (sq *SubmitQueue) validForMergeExt(obj *github.MungeObject, checkStatus boo
 		}
 	}
 
+	if approvedReview, changesRequestedReview, ok := obj.CollectGHReviewStatus(); !ok && (sq.GateGHReviewApproved || sq.GateGHReviewChangesRequested) {
```

Comment:
I think it does not hurt to be explicit:
That way you only call that API when you need it.
if (sq.GateGHReviewApproved || sq.GateGHReviewChangesRequested) {
»       if approvedReview, changesRequestedReview, ok := obj.CollectGHReviewStatus(); !ok {
»       »       sq.SetMergeStatus(obj, ghReviewStateUnclear)
»       »       return false
»       } else if sq.GateGHReviewApproved && len(approvedReview) == 0 { 
»       »       sq.SetMergeStatus(obj, ghReviewApproved)
»       »       return false
»       } else if sq.GateGHReviewChangesRequested && len(changesRequestedReview) > 0 { 
»       »       sq.SetMergeStatus(obj, ghReviewChangesRequested)
»       »       return false
»       }
}

Reply:
make sense